### DISCUSSION TITLE
Deeplink with last page being a modal?

### DISCUSSION THREAD

**Discussion by LeoJHarris** [PrismLibrary/Prism#2689] [General]
Hi all,

I am doing tab selection, deep linking then finally the last page of the deep linking should be a modal page.

This looks like the following with the tab selection omitted:

```csharp
INavigationResult result = await ((CustomViewModelBase)CurrentPage.BindingContext).NavigationService.NavigateAsync(string.Format("{0}/{1}", PageNames.SecurityPage, PageNames.PinCodeSetupPage), useModalNavigation: true).ConfigureAwait(true);

            if (!result.Success)
            {
                result.Exception.WriteToConsole();
            }
```

I want the PinCodeSetupPage to launch as a modal navigation, I setup the useModalNavigation as true but launches the SecurityPage also as a modal. Is there a way to be specific which page launches as a modal in deep linking as the above?

Thanks guys!

**Comment by LeoJHarris** [comment]
Solved it as you often do quickly after asking the question but it might help somebody else.

```csharp
INavigationResult result = await ((CustomViewModelBase)CurrentPage.BindingContext).NavigationService.NavigateAsync(string.Format("{0}/{1}", PageNames.SecurityPage, string.Format("{0}?useModalNavigation=true", PageNames.PinCodeSetupPage))).ConfigureAwait(true);

            if (!result.Success)
            {
                result.Exception.WriteToConsole();
            }
```

**Reply by dansiegel** [reply]
I do not know what you are doing... but I can promise you this is massive code smell

```cs
((CustomViewModelBase)CurrentPage.BindingContext).NavigationService
````
